project/Build.scala

```diff
@@ -1017,6 +1016,30 @@ object Build {
       }.dependsOn(compile in Compile).evaluated
     )
 
+  lazy val `prepare-community-build` = taskKey[Unit]("Publish local the compiler and the sbt plugin. Also store the versions of the published local artefacts in two files, community-build/{dotty-bootstrapped.version,sbt-dotty.sbt}.")
+
+  lazy val `community-build` = project.in(file("community-build")).
+    settings(
+      scalaVersion := scalacVersion,
+      `prepare-community-build` := {
+        (publishLocal in `dotty-sbt-bridge`).value
+        (publishLocal in `dotty-interfaces`).value
+        (publishLocal in `dotty-compiler-bootstrapped`).value
+        (publishLocal in `dotty-library-bootstrapped`).value
+        (publishLocal in `scala-library`).value
+        (publishLocal in `scala-reflect`).value
+        (publishLocal in `dotty-doc-bootstrapped`).value
+        (publishLocal in `dotty-bootstrapped`).value
+        (publishLocal in `sbt-dotty`).value
+        val pluginText = s"""addSbtPlugin("ch.epfl.lamp" % "sbt-dotty" % "$sbtDottyVersion")"""
+        IO.write(baseDirectory.value / "sbt-dotty.sbt", pluginText)
+        IO.write(baseDirectory.value / "dotty-bootstrapped.version", dottyVersion)
+      },
+      (Test / testOnly) := ((Test / testOnly) dependsOn `prepare-community-build`).evaluated,
+      (Test / test    ) := ((Test / test    ) dependsOn `prepare-community-build`).value,
+      libraryDependencies += "com.novocode" % "junit-interface" % "0.11" % Test,
+      testOptions in Test += Tests.Argument(TestFrameworks.JUnit, "-a", "-v")
```

Comment:
You should use `.settings(commonSettings)` to avoid repeating junit settings, etc.